// The Firmament project
// Copyright (c) 2011-2012 Malte Schwarzkopf <malte.schwarzkopf@cl.cam.ac.uk>
//
// Job descriptor protobuf.

package firmament;

import "base/task_desc.proto";

message JobDescriptor {
  enum JobState {
    NEW = 0;
    CREATED = 1;
    RUNNING = 2;
    COMPLETED = 3;
    FAILED = 4;
    ABORTED = 5;
    UNKNOWN = 6;
  }

  required string uuid = 1;
  optional string name = 2;
  optional JobState state = 3;
  required TaskDescriptor root_task = 4;
  repeated bytes output_ids = 5;
}

